### PR TITLE
Improve typing and change eslint config

### PR DESCRIPTION
**Pull request by jakekall**
#28 
- removed all instances of type any
- made a new folder for types and moved RepoFile there
- changed eslint config so that it warns about explicit usage of any
- changed eslint config to match projects current coding style

Interfaces that are used all over the place -> types folder
Interfaces that are used in one class (like Props) -> stay in that class

Wasn't sure what to do with redux types so I left them as they were. Suggestions?

### CODE REVIEW

**Comment by jakekall** [comment]
All types except props are now in src/types.ts

File: frontend/src/types.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+interface GetFilesFailAction {
+    type: typeof GET_FILES_FAIL
+}
+
+export type FileActionTypes = InitGetFiles | GetFilesSuccessAction | GetFilesFailAction
+
+export interface RootState {
+    files: FilesState
+}
```
Comment (ilkkamaksy) [inline frontend/src/types.ts:37]: Oliskohan sittenkin parempi pitää nää reduxin action typesit siellä storessa?

Comment (jakekall) [reply frontend/src/types.ts:37]: Noitahan ei koskaan ilmeisesti tarvita reduxin ulkopuolella eli mun puolesta ne voi hyvin olla storessa.

File: frontend/src/components/MonacoEditor.tsx [hunk 2]
```diff
@@ -1,7 +1,11 @@
 import React from 'react'
 import Editor from '@monaco-editor/react'
 
-const MonacoEditor = ({ content }: any) => {
+interface Props {
```
Comment (sinyman) [inline frontend/src/components/MonacoEditor.tsx:4]: Could this be made more generic and usable in many different components?

File: frontend/src/components/EditView.tsx [hunk 3]
```diff
@@ -1,14 +1,17 @@
 import React from 'react'
 import MonacoEditor from './MonacoEditor'
 import { useSelector } from 'react-redux'
-import { RootState } from '../store/store'
+import { RootState, RepoFile } from '../types'
 
-interface RepoFile {
-    filename: string,
-    content: string,
+interface Props {
```
Comment (sinyman) [inline frontend/src/components/EditView.tsx:6]: Could this be made more generic and usable in many different components?

File: frontend/src/components/MonacoEditor.tsx (hunk 2 above)
Comment (sinyman) [reply frontend/src/components/MonacoEditor.tsx:4]: I am referring to the declaration of the Props-interface outside of the types-directory

File: frontend/src/components/EditView.tsx (hunk 3 above)
Comment (sinyman) [reply frontend/src/components/EditView.tsx:6]: I am referring to the declaration of the Props-interface outside of the types-directory

File: frontend/src/components/EditView.tsx [hunk 4]
```diff
@@ -1,14 +1,17 @@
...
+    location: Location
 }
 
-const EditView = ({ location }: any) => {
+interface Location {
+    search: string
+}
```
Comment (tulma95) [inline frontend/src/components/EditView.tsx:12]: Location type can and should be imported from React-router history:
`import {  Location } from 'history'`

File: frontend/src/components/EditView.tsx (hunk 3 above)
Comment (jakekall) [reply frontend/src/components/EditView.tsx:6]: The way I see it is that the point of having these props interfaces is to define specifically what fields props need to have, so trying to make them generic right away would be pointless. Of course if multiple components end up needing similar props we can make some generic interfaces.

**Comment by jakekall** [comment]
The current idea is:
- types that are used in one file -> defined in that file (refactored to more generic types if there is a reason to)
- redux action types -> stay in the store folder since they are used only by things in the store
- everything else -> src/types file

**Review by tulma95 — APPROVED** [review]
LGTM

**Review by guotin — APPROVED** [review]
Looks good

**Pull request merged**